src/test/java/com/commercetools/sync/products/helpers/VariantReferenceResolverTest.java

```diff
@@ -26,461 +24,418 @@
+        final ObjectNode productReferenceWithRandomId = getProductReferenceWithRandomId();
+        final AttributeDraft productReferenceSetAttribute =
+            getReferenceSetAttributeDraft("foo", productReferenceWithRandomId);
 
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolvePricesReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+        final AttributeDraft textAttribute = AttributeDraft.of("attributeName", "textValue");
 
-        final List<PriceDraft> resolvedBuilderPrices = resolvedBuilder.getPrices();
-        assertThat(resolvedBuilderPrices).isNotEmpty();
-        final PriceDraft resolvedPriceDraft = resolvedBuilderPrices.get(0);
-        assertThat(resolvedPriceDraft).isNotNull();
-        assertThat(resolvedPriceDraft.getCustom()).isNotNull();
-        assertThat(resolvedPriceDraft.getCustom().getType().getId()).isEqualTo("typeId");
-    }
+        final List<AttributeDraft> attributeDrafts =
+            Arrays.asList(productReferenceSetAttribute, textAttribute);
 
-    @Test
-    void resolveAttributesReferences_WithNullAttributes_ShouldNotResolveAttributes() {
-        final ProductVariantDraftBuilder productVariantDraftBuilder = ProductVariantDraftBuilder.of();
 
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolveAttributesReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+        final ProductVariantDraft variantDraft = ProductVariantDraftBuilder.of()
+                                                                           .attributes(attributeDrafts)
+                                                                           .build();
 
-        final List<AttributeDraft> resolvedBuilderAttributes = resolvedBuilder.getAttributes();
-        assertThat(resolvedBuilderAttributes).isNull();
-    }
 
-    @Test
-    void resolveAttributesReferences_WithANullAttribute_ShouldNotResolveAttributes() {
-        final ProductVariantDraftBuilder productVariantDraftBuilder =
-            ProductVariantDraftBuilder.of().attributes((AttributeDraft) null);
+        final ProductVariantDraft resolvedDraft = referenceResolver.resolveReferences(variantDraft)
+                                                                   .toCompletableFuture().join();
 
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolveAttributesReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+        final List<PriceDraft> resolvedDraftPrices = resolvedDraft.getPrices();
+        assertThat(resolvedDraftPrices).isNull();
 
-        final List<AttributeDraft> resolvedBuilderAttributes = resolvedBuilder.getAttributes();
-        assertThat(resolvedBuilderAttributes).isEmpty();
+        final List<AttributeDraft> resolvedBuilderAttributes = resolvedDraft.getAttributes();
```

Comment:
could you please add space between asserts to make it easy to read.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/442/commits/85aa403766af91f5ca22ee8009a75dcb6adadb7b